tests/gateway/test_stripe.py

```diff
@@ -0,0 +1,648 @@
+from saleor.payment.gateways.stripe import (
+    _create_transaction, _get_client, _get_error_response_from_exc, authorize,
+    capture, charge, get_amount_for_stripe, get_amount_from_stripe,
+    get_client_token, get_currency_for_stripe, get_currency_from_stripe,
+    get_form_class, refund, void)
+from saleor.payment.gateways.stripe.errors import (
+    ORDER_NOT_AUTHORIZED, ORDER_NOT_CHARGED)
+from saleor.payment.gateways.stripe.forms import (
+    StripeCheckoutWidget, StripePaymentModalForm)
+
+TRANSACTION_AMOUNT = Decimal(42.42)
+TRANSACTION_REFUND_AMOUNT = Decimal(24.24)
+TRANSACTION_CURRENCY = 'USD'
+TRANSACTION_TOKEN = 'fake-stripe-id'
+FAKE_TOKEN = 'fake-token'
+ERROR_MESSAGE = 'error-message'
+
+
+@pytest.fixture()
+def gateway_params():
+    return {
+        'public_key': 'public',
+        'secret_key': 'secret',
+        'store_name': 'Saleor',
+        'store_image': 'image.gif',
+        'prefill': True,
+        'remember_me': True,
+        'local': 'auto',
+        'enable_billing_address': False,
+        'enable_shipping_address': False}
+
+
+@pytest.fixture()
+def client_token():
+    return FAKE_TOKEN
+
+
+@pytest.fixture()
+def stripe_payment(payment_dummy):
+    payment_dummy.total = TRANSACTION_AMOUNT
+    payment_dummy.currency = TRANSACTION_CURRENCY
+    return payment_dummy
+
+
+@pytest.fixture()
+def stripe_authorized_payment(stripe_payment):
+    stripe_payment.charge_status = ChargeStatus.NOT_CHARGED
+    stripe_payment.save(update_fields=['charge_status'])
+
+    stripe_payment.transactions.create(
+        amount=stripe_payment.total,
+        kind=TransactionKind.AUTH,
+        gateway_response={},
+        is_success=True)
+    return stripe_payment
+
+
+@pytest.fixture()
+def stripe_captured_payment(stripe_payment):
+    stripe_payment.captured_amount = stripe_payment.total
+    stripe_payment.charge_status = ChargeStatus.CHARGED
+    stripe_payment.save(update_fields=['captured_amount', 'charge_status'])
+
+    stripe_payment.transactions.create(
+        amount=stripe_payment.total,
+        kind=TransactionKind.CAPTURE,
+        gateway_response={},
+        is_success=True)
+    return stripe_payment
+
+
+@pytest.fixture()
+def stripe_charged_payment(stripe_payment):
+    stripe_payment.captured_amount = stripe_payment.total
+    stripe_payment.charge_status = ChargeStatus.CHARGED
+    stripe_payment.save(update_fields=['captured_amount', 'charge_status'])
+
+    stripe_payment.transactions.create(
+        amount=stripe_payment.total,
+        kind=TransactionKind.CHARGE,
+        gateway_response={},
+        is_success=True)
+    return stripe_payment
+
+
+@pytest.fixture()
+def stripe_charge_success_response():
+    return {
+        'id': TRANSACTION_TOKEN,
+        'amount': get_amount_for_stripe(
+            TRANSACTION_AMOUNT, TRANSACTION_CURRENCY),
+        "amount_refunded": 0,
+        'currency': get_currency_for_stripe(TRANSACTION_CURRENCY),
+        "status": "succeeded"}
```

Comment:
Accidental double quotes here (and a bit above)

Reply:
Thanks for the careful review! :smile: